### PR TITLE
add a middleware to record the status code return by an http server (…

### PR DESCRIPTION
**Pull request by JessicaGreben**
…#453)

### CODE REVIEW

**Review by fishy — APPROVED** [review]
please use rebase instead of squash to merge this.

File: httpbp/middlewares.go [hunk 1]
```diff
@@ -246,3 +255,133 @@ func SupportedMethods(method string, additional ...string) Middleware {
 		}
 	}
 }
+
+// recoverPanic recovers from any panics, logs them, and sets the returned error
+// to a generic 500 error. recoverPanic is always the last middleware in the
+// middleware chain, so it is the first one when returning which lets the error
+// bubble up into other middlewares. Since it is always added to the middleware
+// chain is a specific position, it is not exported.
+func recoverPanic(name string, next HandlerFunc) HandlerFunc {
```
Comment (pacejackson) [inline httpbp/middlewares.go:264]: This should be in a separate commit, I think you should cherry pick multiple commits and do "rebase and merge" rather than squashing them.

**Review by pacejackson — CHANGES_REQUESTED** [review]

File: httpbp/middlewares.go (hunk 1 above)
Comment (pacejackson) [reply httpbp/middlewares.go:264]: Also, it looks like the other parts of the change that added this are missing from this PR, you probably want to cherry-pick 

91c387c4c9ba9f4d468cea5cbfa2533e94d0f323

and 

56f88638113fea6bf488b288a0eedfc593bf9673

Comment (kylelemons) [reply httpbp/middlewares.go:264]: this is why we do merges 😅

**Comment by kylelemons** [comment]
🏃‍♂️ since it's @pacejackson commit, so deferring to him :)

**Comment by fishy** [comment]
You likely also want to cherry-pick 846e891, that fixes a flaky test that can affect the CI runs of the PRs on the prometheus branch.

**Comment by fishy** [comment]
and also 43a52d3.

**Review by fishy — APPROVED** [review]

**Review by pacejackson — APPROVED** [review]
